### ISSUE TITLE
Remove separately maintained ro_import and dot-definitions from uberon-merged.owl

### ISSUE THREAD

**Issue by gouttegd** [obophenotype/uberon#2967]
Even though Uberon has switched to the base merging mechanism several months ago already, it is still actively using the individual `ro_import.owl` module, which was last updated in January.

This is because of this step early in the pipeline:

```
$(OWLSRC): tmp/uberon-merged.owl $(COMPONENTSDIR)/disjoint_union_over.ofn $(REPORTDIR)/$(SRC)-gocheck $(REPORTDIR)/$(SRC)-iconv $(SCRIPTSDIR)/expand-dbxref-literals.pl
	echo "STRONG WARNING: issues/contributor.owl needs to be manually updated."
	$(OWLTOOLS) --no-logging $< $(COMPONENTSDIR)/disjoint_union_over.ofn issues/contributor.owl --merge-support-ontologies -o $@ && $(SCRIPTSDIR)/expand-dbxref-literals.pl $@ > $@.tmp
	# The previous step seems to be necessary because somehow the expand-dbxref-literals script expects the owtools output.. No idea why
	$(ROBOT) expand -i $@.tmp --no-expand-term http://purl.obolibrary.org/obo/RO_0002175 \
		query \
			--update $(SPARQLDIR)/remove_axioms.ru \
			--update $(SPARQLDIR)/delete-definition-dot.ru \
		merge -i $(IMPORTDIR)/ro_import.owl -o $@
```

Among other things, this step drops every single axioms relating to RO (`--update $(SPARQLDIR)/remove_axioms.ru`), then re-imports them from the individual `ro_import.owl` module (`merge -i $(IMPORTDIR)/ro_import.owl`).

As a result, all the RO axioms coming from the (up-to-date) `merge_import.owl` module are discarded, and replaced by the RO axioms coming from the (stale) `ro_import.owl` module. Among other consequences, it means that several bogus range/domain constraints that had been fixed in RO in the recent months are creeping back in.

The rationale for this “remove-then-reimport” dance was to “remove RO labels to avoid duplicates then adds them back to ensure they are canonical”. I believe after the switch to base merging this is now completely unnecessary, in addition to being harmful.

**Comment by gouttegd** [comment]
Of note, the other things done in that step seems equally dubious to me.

1) The `expand-dbxref-literals.pl` script does… nothing. It is supposed to expand CURIEs in some `oboInOwl:` annotations (_not_ `oboInOwl:hasDbXref` annotations) into IRIs, but it does not actually find a single CURIE to expand.

2) The `delete-definition-dot.ru` removes definition annotations whose value is a single dot. However, it does not remove any annotation on the definition itself, which then remain lingering around attached to nothing.

That is, that step transforms this:

```
AnnotationAssertion(Annotation(oboInOwl:hasDbXref "BIRNLEX:1734") obo:IAO_0000115 obo:UBERON_0006100 ".")
```

(a definition that is a single dot with a `BIRNLEX:1734` cross-reference attached to it) into this:

```
AnnotationAssertion(oboInOwl:hasDbXref _:genid2147650234 "BIRNLEX:1734") 
```

(a `BIRNLEX:1734` cross-reference attached to a blank node).

I fail to see how this is useful. Either we should remove the dot definitions cleanly (removing any associated annotations), or we should leave them alone.

**Comment by anitacaron** [comment]
Thanks for noticing that, Damion. You're right. Now that we have all imports merged, we don't need this part anymore.
I'm removing the dot definitions from the edit file so we can remove this step soon.

**Comment by gouttegd** [comment]
OK, then I’ll prepare a PR to remove it. :)

**Comment by matentzn** [comment]
A sigh of relief going through my heart seeing 4 sane eyes on this subject! THANKS!

**Comment by gouttegd** [comment]
Bold of you to assume I am sane… :D

**Comment by anitacaron** [comment]
As discussed in the linked PR (the script `expand-dbxref-literals.pl` needs to be fixed instead of being completely removed in this pipeline), could you edit the scope of this issue to only:

(1) remove the `ro_import.owl` merge
(2) remove the sparql query to remove the "dot" definitions

**Comment by gouttegd** [comment]
> the script expand-dbxref-literals.pl needs to be fixed instead of being completely removed in this pipeline

Not until I get positive confirmation (and not merely supposition) that what this script is supposed to do (and has not been doing for the past six months) is an actively desired behaviour. Then I’ll do it happily.

But until then, I consider this script to be part of the same kind of old hacks as the “remove-RO-axioms-then-reimport-them”, and removing its invocation is completely in scope for this issue.

**Comment by gouttegd** [comment]
And by the way, if/when someones does ask for CURIE expansion to be restored, they can use that opportunity to state clearly what should be expanded and where, exactly.

The current script, back when it worked, only expanded CURIEs in `oboInOwl:source` and `oboInOwl:hasDbXref` annotations and only expanded a handful of prefixes. Was there a reason for that? Were the other prefixes left unexpanded on purpose? Should we stick to expand only those prefixes in those annotations? All things I’d very much like to know before I do anything.

**Comment by matentzn** [comment]
Moved the expand-dbxref-literal discussion to a separate issue: https://github.com/obophenotype/uberon/issues/2974

**Comment by matentzn** [comment]
@gouttegd I remember putting that expand-curie thing there in the past, and I see why it does not work (it only works on OBO format :P), but it has nothing to do with the RO import. I moved it out not because it is entirely separate, but because I can confidently confirm that ro imports and dot definition can go now.. I have informed @cmungall of our intention to drop the expand uris script as well, and instructed him to warn us in case he needs to. I personally I agree with your judgement on the matter though: I fail to see the purpose of any it.

**Comment by gouttegd** [comment]
> I remember putting that expand-curie thing there in the past, and I see why it does not work (it only works on OBO format :P)

Are you sure you’re not talking about something else? The `expand-dbxref-literals.pl` script is _not_ intended to be used on OBO files. It does not work not because it is used on a RDF/XML file (that’s the format it is supposed to be used on), but because it expects OWL statements to be serialised into RDF/XML in a precise way, which is no longer the way they are serialised since OWL API 4.5.25.

**Comment by matentzn** [comment]
Ringing some bells now that you say this again.. Is there a reason you want to couple these issues? I would have thought quick focussed iterations is more your style then long composite iterations? I mean, we can wait for Chris, maybe he gives the green light - or we reconceptualise issue #2974 to reinstate the script if it is needed for something we don't see here, with the argument that right now, its broken.. No strong feelings either way, just sharing Anitas principle for separation of concerns.

**Comment by gouttegd** [comment]
And by the way:

> I personally I agree with your judgement on the matter though: I fail to see the purpose of any it.

I am _not_ making a judgment on the usefulness of CURIE expansion. Frankly, I couldn’t care less about CURIE expansion. I am making a judgment about our tendency to keep old crufts around without ever pausing to consider whether it’s needed or not, because we are too afraid to make changes.

> Is there a reason you want to couple these issues?

It’s a single issue: the pre-processing step is full of leftover craps. The most important crap is the ”remove-RO-axioms-then-reimport-them”, but the useless invocation of a script that in its present state does nothing is part of the same problem.

PR #2968 is all about removing that crap.

The question of whether CURIEs should be expanded or not has nothing to do with it. Right now, they are _not_ expanded, and my PR does not change anything to that.

If someone shows up to say, “hey, I actually need CURIEs to be expanded”, _that_ would be another issue, which can be fixed in another PR.

> No strong feelings either way, just sharing Anitas principle for separation of concerns.

It’s precisely _because_ of that principle that I refused to fix the CURIE expansion script in PR #2968, as she requested.

**Comment by matentzn** [comment]
Lovely, thanks for the clarifications! We are on the same page. You got my vote to proceed! I added #2974 is `very high` on the tech team board to not loose sight of it.
